pep-0622.rst

```diff
@@ -22,88 +22,139 @@ Resolution:
 
 - ``[first, second, *rest]``
 - ``Point2d(x, 0)``
 - ``{"name": "Bruce", "age": age}``
 - ``42``
 
-The above look like examples of object construction. A constructor
-takes some values as parameters and builds an object from those
-components. But as a pattern the above mean the inverse operation of
-construction, which we call *destructuring*: it takes a subject value
-and extracts its components. The syntactic similarity between
-construction and destructuring is intentional and follows the existing
-Pythonic style which makes assignment targets (write contexts) look
-like expressions (read contexts). Pattern matching never creates
-objects, in the same way that ``[a, b] = my_list`` doesn't create a
+The above expressions may look like examples of object construction
+with a constructor which takes some values as parameters and
+builds an object from those components.
+
+When viewed as a pattern, the above patterns mean the inverse operation of
+construction, which we call **destructuring**. **Destructuring** takes a subject value
+and extracts its components.
+
+The syntactic similarity between object construction and destructuring is
+intentional. It also follows the existing
+Pythonic style of contexts which makes assignment targets (write contexts) look
+like expressions (read contexts).
+
+Pattern matching never creates objects. This is in the same way that
+``[a, b] = my_list`` doesn't create a
 new ``[a, b]`` list, nor reads the values of ``a`` and ``b``.
 
-The intuition we are trying to build in users as they learn this is
-that matching a pattern to a subject binds the free variables (if any)
-to subject components in a way that reflects the original
-subject when read as an expression. During this process,
-the structure of the pattern may not fit the subject, in which case
-the matching *fails*. For example, matching the pattern ``Point2d(x,
-0)`` to the subject ``Point2d(3, 0)`` successfully matches and binds
-``x`` to ``3``. However, if the subject is ``[3, 0]`` the match fails
-because a ``list`` is not a ``Point2d``. And if the subject is
-``Point2D(3, 3)`` the match fails because its second coordinate is not
-``0``.
-
-The ``match`` statement tries to match each of the
-patterns in its ``case`` clauses with a single subject. At the first
-successful match, the variables in the pattern are assigned and a
-corresponding block is executed. Each of the multiple branches of this
-conditional statement can also have a boolean condition as a *guard*.
-
-Here's an example of a match statement, used to define a function
-building 3D points that can accept as input either tuples of size 2 or
-3, or existing (2D or 3D) points::
+
+Matching process
+----------------
+
+..  **Reword**
+    The intuition we are trying to build in users as they learn this is
+    that matching a pattern to a subject binds the free variables (if any)
+    to subject components in a way that reflects the original
+    subject when read as an expression.
```

Comment:
Matching a pattern to a subject binds free variables in the pattern to subject components (i.e. sequence elements, mapping values, object attributes). This is done in the way that would recreate an object similar to the original if fed back into an expression that looks like the pattern. Once again, this is similar to sequence unpacking where the target ``a, b`` that was assigned ``my_list`` can be evaluated as an expression ``a, b`` which produces something similar to ``my_list``.

Reply:
Maybe @willingc can rewrite your paragraph to her liking? Personally I would rather land this with her note in it than wait for perfection. PRs are cheap (at least in this repo :-).